### DISCUSSION TITLE
Many chromium helper processes remain

### DISCUSSION THREAD

**Discussion by cha0ran** [spatie/browsershot#541] [Q&A]
When I am developing under mac, I found that there are a lot of Chromium Helper (GPU & Renderer) processes left in the background, how should I avoid or close them?

<img width="500" alt="WX20210810-100002@2x" src="https://user-images.githubusercontent.com/13059055/128797209-b77a8db9-072d-4d2f-814b-92043a1c265e.png">

<img width="1279" alt="WX20210810-100451@2x" src="https://user-images.githubusercontent.com/13059055/128797450-64c6f681-fbcd-4955-8ad4-83489a4f4ed1.png">

**Comment by Nutickets** [comment]
We are experiencing this same issue but in our production servers and it's causing our CPU to sit at 90%+

![Screenshot 2022-01-26 at 19 43 30](https://user-images.githubusercontent.com/57409655/151254132-8938cba0-c0f1-496b-bfc5-5f551c4f2593.png)

What I've discovered is that this seems to be only an issue from a cold start. If I reboot the server and render a PDF using browsershot, a dozen or so chrome processes spawn and get stuck forever. If I kill those processes manually and then re-render the PDF, it works fine and the processes don't stick around. But if I don't kill the processes, then they stay constantly active and the PDF generation is a bit hit and miss and the whole server becomes sluggish

Any help would be much appreciated

**Comment by vatandoost** [comment]
These arguments of chromium fixed the issue for me.

```
$browsershot->addChromiumArguments([
   'no-zygote',
   'single-process',
]);
```

https://source.chromium.org/chromium/chromium/src/+/main:content/public/common/content_switches.cc;drc=b8152085aa045fe26c428aceaf99b5c83d53caac;l=572

**Reply by kriiv** [reply]
Curious if this would affect the ability for concurrent instances to run on multiple cores in production?

**Comment by Daniel-FDS** [comment]
I am experiencing the same think on browsershot version 3. Is there a way to fix this?

![image](https://github.com/user-attachments/assets/29eb2ce4-0d7f-4995-ab95-53991e70a7c3)

**Reply by socrec** [reply]
Same thing here
